src/Compilers/Core/Portable/DiagnosticAnalyzer/ShadowCopyAnalyzerAssemblyLoader.cs

```diff
@@ -11,60 +11,69 @@
 using Roslyn.Utilities;
 using System.Collections.Immutable;
 using System.Reflection;
-
-#if NET
-using System.Runtime.Loader;
-#endif
+using System.Globalization;
 
 namespace Microsoft.CodeAnalysis
 {
-    internal sealed class ShadowCopyAnalyzerAssemblyLoader : AnalyzerAssemblyLoader
+    internal sealed class ShadowCopyAnalyzerPathResolver : IAnalyzerPathResolver
     {
         /// <summary>
         /// The base directory for shadow copies. Each instance of
-        /// <see cref="ShadowCopyAnalyzerAssemblyLoader"/> gets its own
+        /// <see cref="ShadowCopyAnalyzerPathResolver"/> gets its own
         /// subdirectory under this directory. This is also the starting point
         /// for scavenge operations.
         /// </summary>
         private readonly string _baseDirectory;
 
+        /// <summary>
+        /// The name of the shadow directory. This will be the same name as used for the <see cref="_mutex"/>
+        /// field. That allows other instances of this type to coordinate on active and inactive
+        /// shadow copy directories.
+        /// </summary>
+        private readonly string _shadowDirectoryName;
+
+        private readonly string _shadowDirectory;
```

Comment:
perhaps _shadowDirectoryPath

Reply:
I try to use `Path` or `Directory` as a suffix for paths and `Name` when it's a file name. I used to say `DirectoryPath` but names got too windy for some people.